lib/ansible/modules/cloud/amazon/ec2_group.py

```diff
@@ -284,6 +295,27 @@ def get_target_from_rule(module, ec2, rule, name, group, groups, vpc_id):
     return group_id, ip, target_group_created
 
 
+def ensure_tags(ec2, group, tags, changed, module):
+    try:
+        resource_id = group.id
+        cur_tags = group.tags
+        if cur_tags == tags:
```

Comment:
The issue with floating point stuff is that it will drop trailing digits and/or lose precision. A tag value of 1.000 for example would be interpreted by Python as 1.0 unless properly quoted.

Either way - there's no consistent validation for tags across modules. I think I saw a PR float through recently that would add tag handling code to ec2.py, at which point all the modules should be converted over to use it for consistency's sake - but as of right now the issues we're talking about are widespread.